controllers/util/k8sutil.go

```diff
@@ -60,6 +60,7 @@ const (
 	// This is the name for the FieldIndex
 	OwnerUIDIndexName   = "ownerUID"
 	AnnotationIndexName = "annotation"
+	UIDIndexName        = "uid"
```

Comment:
let's have a suffix to the index as plain uid is actually a field in every resource.